swupd/manifest.go

```diff
@@ -480,11 +487,52 @@ func (m *Manifest) linkPeersAndChange(oldManifest *Manifest, c config, minVersio
 func (m *Manifest) newDeleted(df *File) {
 	df.Version = m.Header.Version
 	df.Status = StatusDeleted
+	df.Type = TypeUnset
```

Comment:
I know we need this for compatibility, but I think it is wrong. I don't see why a deleted {file,directory.link} is no longer a {file,directory.link} for comparison purposes. It does mean that we should probably be checking for df.Status in more places.